build-config/make/mtdutils.make

```diff
@@ -36,17 +38,24 @@ mtdutils-download: $(MTDUTILS_DOWNLOAD_STAMP)
 $(MTDUTILS_DOWNLOAD_STAMP): $(PROJECT_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Getting upstream mtdutils ===="
-	$(Q) $(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
-		$(MTDUTILS_COMMIT).tar.gz $(MTDUTILS_TARBALL_URLS)
-	$(Q) cd $(DOWNLOADDIR) && ln -fs $(MTDUTILS_COMMIT).tar.gz $(MTDUTILS_TARBALL)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(MTDUTILS_VERSION)'"<="'$(MTDUTILS_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+		$(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
+			$(MTDUTILS_COMMIT).tar.gz $(MTDUTILS_TARBALL_URLS);\
+		cd $(DOWNLOADDIR) && ln -fs $(MTDUTILS_COMMIT).tar.gz $(MTDUTILS_TARBALL);\
+		 fi
 	$(Q) touch $@
 
 SOURCE += $(MTDUTILS_SOURCE_STAMP)
 mtdutils-source: $(MTDUTILS_SOURCE_STAMP)
 $(MTDUTILS_SOURCE_STAMP): $(TREE_STAMP) | $(MTDUTILS_DOWNLOAD_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Extracting upstream mtdutils ===="
-	$(Q) $(SCRIPTDIR)/extract-package $(MTDUTILS_BUILD_DIR) $(DOWNLOADDIR)/$(MTDUTILS_TARBALL)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(MTDUTILS_VERSION)'"<="'$(MTDUTILS_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/extract-package $(MTDUTILS_BUILD_DIR) $(DOWNLOADDIR)/$(MTDUTILS_TARBALL); \
+		 else \
+			git clone git://git.infradead.org/mtd-utils.git \
```

Comment:
Same comments about cloning master -- this will bite us eventually.  Let's pick a release (a commit) and make a tarball of it.  Same comments I made on the crosstool-ng.